Statistiscal Analysis

Chi-square Tests

In [1]:
# import necessary libraries 
import pandas as pd
import numpy as np 
import seaborn as sb 
from scipy import stats

In [2]:
# load the dataset 
df = sb.load_dataset("titanic") 
df.head()

survived  pclass     sex   age  sibsp  parch     fare embarked  class  \
0         0       3    male  22.0      1      0   7.2500        S  Third   
1         1       1  female  38.0      1      0  71.2833        C  First   
2         1       3  female  26.0      0      0   7.9250        S  Third   
3         1       1  female  35.0      1      0  53.1000        S  First   
4         0       3    male  35.0      0      0   8.0500        S  Third   

     who  adult_male deck  embark_town alive  alone  
0    man        True  NaN  Southampton    no  False  
1  woman       False    C    Cherbourg   yes  False  
2  woman       False  NaN  Southampton   yes   True  
3  woman       False    C  Southampton   yes  False  
4    man        True  NaN  Southampton    no   True

Null Hypothesis : There is no significant association between gender ('Sex') and survival('Survived') on the Titanic. This means any observed difference in survival rates between genders in the dataset are due to chance and not any kind of underlying relationship.

Alternate Hypothesis : There is a significant association between gender ('Sex') and survival('Survived') on the Titanic. This implies that the difference in survival rates are not just due to chance but are influenced by the passengers' gender.

In [5]:
# Contingency table for 'sex' and 'survived'
contigency_table = pd.crosstab(df['survived'], df['sex'])
contigency_table

sex       female  male
survived              
0             81   468
1            233   109

In [ ]:
# Perform Chi-Sqaured test 
chi2,p,dof,exp=stats.chi2_contingency(contigency_table)
# Print the results
print("Chi-Squared Test Results:") 
print("Chi2: ",chi2) 
print("p-value: ",p) 
print("Degree of Freedom ",dof) 
print("Expected Frequencies: \n", exp) 

Chi-Squared Test Results:
Chi2:  260.71702016732104
p-value:  1.1973570627755645e-58
Degree of Freedom  1
Expected Frequencies: 
 [[193.47474747 355.52525253]
 [120.52525253 221.47474747]]


The above chi-sqaured test gave a p-value of 1.1973570627755645e-58(1.1973570627755645 * 10^-58) which is far less than 0.05 confidence interval. So we reject Null hypothes and accept alternate hypothesis meaning there is significant difference between the gender and survived column.